### DISCUSSION TITLE
Piping output from one command into another command

### DISCUSSION THREAD

**Discussion by serbanrobu** [tokio-rs/tokio#3447] [Q&A] [answered]
It's there a way of piping the output of a command into another? I'm trying to make this example working. But the `stdin` method expect a type `T: Into<std::process::Stdio>` as parameter. The following example would work if I used `std::process::Command`.
```rust
let child = tokio::process::Command::new("ls")
    .stdout(std::process::Stdio::piped())
    .spawn()?;
tokio::process::Command::new("grep")
    .arg("a")
    .stdin(child.stdout.unwrap())
    .spawn()?;
```
I get the following error:
```
the trait bound `Stdio: From<tokio::process::ChildStdout>` is not satisfied [...]
```

**Comment by ipetkov** [comment, marked as answer]
Hi @serbanrobu, thanks for bringing this up! As things stand today there isn't an ergonomic way to perform the conversion, but I've opened #3466 which should make this easier to perform the conversion.

In the meantime, you could try getting yourself unblocked by manually creating an anonymous pipe and passing in the write/read ends  into the child processes (tokio will not capture and manage the child's stdio handles if they are explicitly set to a particular file descriptor).

Hope this helps, but don't hesitate to ask if you have other questions!

**Reply by serbanrobu** [reply]
Thank you!
